lib/fog/aws/requests/dynamodb/query.rb

```diff
@@ -25,15 +23,17 @@ class Real
         #     * 'Count'<~Integer> - number of items in response
         #     * 'Items'<~Array> - array of items returned
         #     * 'LastEvaluatedKey'<~Hash> - last key scanned, can be passed to ExclusiveStartKey for pagination
-        def query(table_name, hash_key, options = {})
+        #
+        # See DynamoDB Documentation: http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html
+        #
+        def query(table_name, options = {})
```

Comment:
I'm not super familiar with the new stuff here, but is there any way we could take either format here? ie if hash_key is supplied, warn about deprecation but then merge it appropriately into the options and continue? If so, would be great to avoid the backwards-incompatibility here.